app/models/assessment.rb

```diff
@@ -46,7 +46,8 @@ class Assessment < ApplicationRecord
   # Constants
   ORDERING = "due_at ASC, name ASC".freeze
   RELEASED = "start_at < ?".freeze
-
+  VALID_NAME_REGEX = /^[A-Za-z][A-Za-z0-9_-]*$/.freeze
+  VALID_NAME_SANITIZER_REGEX = /^[^A-Za-z]*([^A-Za-z]*[A-Za-z0-9_-]+)/.freeze
```

Comment:
Nit: the second `[^A-Za-z]*` seems redundant, could simplify to `/^[^A-Za-z]*([A-Za-z0-9_-]+)/`